src/pointer/ptr.rs

```diff
@@ -447,6 +449,65 @@ mod _conversions {
             unsafe { raw.as_ref() }
         }
     }
+
+    /// `&'a MaybeUninit<T>` → `Ptr<'a, T>`
+    impl<'a, T> Ptr<'a, T, (invariant::Shared, invariant::Aligned, invariant::AnyValidity)>
+    where
+        T: 'a,
+    {
+        /// Constructs a `Ptr` from a shared reference to a [`MaybeUninit<T>`].
+        ///
+        /// [`MaybeUninit<T>`]: MaybeUninit
+        #[doc(hidden)]
+        #[inline]
+        pub fn from_maybe_uninit_ref(ptr: &'a MaybeUninit<T>) -> Self {
+            let muptr = core::ptr::NonNull::from(ptr);
+            let ptr = muptr.cast::<T>();
+            // SAFETY:
+            // 0. `muptr`, by invariant on `&'a T`, is derived from some valid
+            //    Rust allocation, `A`. `ptr` is as well because `.cast()`
```

Comment:
I assume you don't just consistently shadow `ptr` (and instead introduce `muptr`) because you want to distinguish between these bindings in our safety comments. If so, consider that we have *two* bindings here named `ptr`; perhaps we should rename one of those, too.

Reply:
Done.

Reply:
Done.